ecl/agentexec/agentexec.cpp

```diff
@@ -253,25 +267,28 @@ class WaitThread : public CInterfaceOf<IPooledThread>
             bool useChildProcesses = compConfig->getPropBool("@useChildProcesses");
             if (isContainerized() && !useChildProcesses)
             {
-                std::list<std::pair<std::string, std::string>> params = { };
-                if (compConfig->getPropBool("@useThorQueue", true))
-                    params.push_back({ "queue", queue.get() });
-                StringBuffer jobName(wuid);
-                if (isThorAgent)
+                if (!owner.lingerQueue || !queueJobIfQueueWaiting(owner.lingerQueue, item, 10000, 1000))
```

Comment:
Worth exposing the two timeouts as named constants to improve readability.  There is a possibility they might be configured in the future.